File: src/Samples/NHaml.Samples.Mvc.Boo/Models/NorthwindDataContext.cs
using System.Collections.Generic;
using System.Linq;

namespace NHaml.Samples.Mvc.Boo.Models
{
    public partial class NorthwindDataContext
    {
        // Retrieve All Category Objects

        public List<Category> GetCategories()
        {
            return Categories.ToList();
        }

        // Retrieve all Suppliers

        public List<Supplier> GetSuppliers()
        {
            return Suppliers.ToList();
        }

        // Retrieve a Specific Category

        public Category GetCategoryById( int categoryId )
        {
            return Categories.First( c => c.CategoryID == categoryId );
        }

        // Retrieve a specific Product

        public Product GetProductById( int productId )
        {
            return Products.Single( p => p.ProductId == productId );
        }

        // Add a New Product

        public void AddProduct( Product product )
        {
            Products.InsertOnSubmit( product );
        }
    }
}

File: src/Samples/NHaml.Samples.Mvc.VisualBasic/Models/NorthwindDataContext.vb
Imports System.Collections.Generic
Imports System.Linq

Namespace Models
    Partial Public Class NorthwindDataContext
        ' Retrieve All Category Objects

        Public Function GetCategories() As List(Of Category)
            Return Categories.ToList()
        End Function

        ' Retrieve all Suppliers

        Public Function GetSuppliers() As List(Of Supplier)
            Return Suppliers.ToList()
        End Function

        ' Retrieve a Specific Category

        Public Function GetCategoryById(ByVal categoryId As Integer) As Category
            Return Categories.First(Function(c) c.CategoryID = categoryId)
        End Function

        ' Retrieve a specific Product

        Public Function GetProductById(ByVal productId As Integer) As Product
            Return Products.Single(Function(p) p.ProductID = productId)
        End Function

        ' Add a New Product

        Public Sub AddProduct(ByVal product As Product)
            Products.InsertOnSubmit(product)
        End Sub
    End Class
End Namespace